src/spline2/MultiBsplineData.hpp

```diff
@@ -31,9 +31,24 @@ namespace spline2
 template<typename T, typename TRESIDUAL>
 inline void getSplineBound(T x, TRESIDUAL& dx, int& ind, int ng)
 {
-  T ipart;
-  dx=std::modf(x,&ipart);
-  ind = std::min(std::max(int(0),static_cast<int>(ipart)),ng);
+  // lower bound
+  if (x < 0)
+  {
+    ind = 0;
+    dx = T(0);
```

Comment:
So can this spline be std::complex or some other type that needs to be explicitly constructed with 0 to be zero?  The T(0) implies that.

Reply:
spline is real only but can be float or double.
The compile should optimized it out. T is plain-old type and the compiler should optimize it out.